rsnapshot-program.pl

```diff
@@ -5254,6 +5254,19 @@ sub rm_rf {
 	# make sure we were passed an argument
 	if (!defined($path)) { return (0); }
 
+	# added by Matthew Jurgens as part of the fix for rm -rf failing when the path contains ./
+	# make sure $path does not contain ./ at the end since this makes rm -rf fail
+	# however the side effect of doing this is that we also remove the directory so we will have to create it later
+	# we could have added a * at the end so we had something like rm -rf PATH/* but probably safer not to use the *
+	$path=~s/\.\/$//;
+	# in order to recreate the directory we will want to take a look at it before we remove it
```

Comment:
This has caused a regression: with lazy deletes empty _delete.xxxx directories are left in place after successful execution. TBH I don't get why this isn't a one liner. Why do we need to restore the directory?